qiskit_optimization/algorithms/optimization_algorithm.py

```diff
@@ -551,10 +552,9 @@ def generate_solution(bitstr, qubo, probability):
 
         solutions = []
         if isinstance(eigenvector, dict):
-            all_counts = sum(eigenvector.values())
+            eigenvector = {bitstr: val ** 2 for (bitstr, val) in eigenvector.items()}
```

Comment:
Instead of overwriting `eigenvector`, saving as `probabilities` seems more consistent with the case `isinstance(eigenvector, np.ndarray)`. What do you think?
```
            probabilities = {bitstr: val ** 2 for (bitstr, val) in eigenvector.items()}
```